transport/https.go

```diff
@@ -0,0 +1,62 @@
+package transport
+
+import (
+	"bytes"
+	"encoding/base64"
+	"io"
+	"io/ioutil"
+	"net/http"
+)
+
+func NewHttpsBase64Transport() Transport {
+	return &HttpsBase64Transport{
+		httpClient: http.DefaultClient,
+	}
+}
+
+type HttpsBase64Transport struct {
+	httpClient *http.Client
+}
+
+func (h *HttpsBase64Transport) Do(request *Request) (*Response, error) {
+	var buf bytes.Buffer
+	encodingWriter := base64.NewEncoder(base64.StdEncoding, &buf)
+	_, err := io.Copy(encodingWriter, request.Body)
+	if err != nil {
+		return nil, err
+	}
+	encodingWriter.Close()
+	httpResponse, err := h.httpClient.Post(request.URL, "application/vnd.hbci", &buf)
+	if err != nil {
+		return nil, err
+	}
+	var reader io.Reader
+	if httpResponse.StatusCode == http.StatusOK {
+		reader = base64.NewDecoder(base64.StdEncoding, httpResponse.Body)
+	} else {
+		reader = httpResponse.Body
+	}
+	return &Response{Body: ioutil.NopCloser(reader), Request: request}, nil
+}
+
+func NewHttpsTransport() Transport {
```

Comment:
two cents in terms of naming:

1. the package is called `transport`, so maybe just calling this `New` would be more appropriate, and moving the entire package into `transport/https`.
2. if the name contains `https`, it should be `NewHTTPSTransport`, as it's an abbreviation. 

also, comment are rare these days :)